stdlib/src/builtin/builtin_slice.mojo

```diff
@@ -46,7 +46,7 @@ struct Slice(Stringable, EqualityComparable):
     """The starting index of the slice."""
     var end: OptionalReg[Int]
     """The end index of the slice."""
-    var step: Int
```

Comment:
**Question** I wonder if it would be possible to separate out the change to use `Optional` for the `step` into a follow-up PR?

The performance characteristics of these slice types are somewhat sensitive, so it would be nice to de-risk the valuable formatting improvements here, so that we can evaluate the `step` type change on its own.

Things like that will also get easier as our benchmarking story improves 🙂

Reply:
I think that's fair, I can certainly do that